### PR TITLE
Dev guide: mention field types for manifest, along with examples edited to the new format

### LINKED ISSUE #8593: Add-ons Manager: add-ons with commas and other punctuations not surrounded by quotes in manifest/summary field shows Python-like text
Hi,

Found while troubleshooting an issue with Add-on Updater add-on:

### Steps to reproduce:
1. Install Vocalizer Expressive English Compact add-on.
2. Restart NvDA when prompted.
3. Open Add-ons Manager and look for the new add-on.

### Actual behavior:
The name shown resembles Python's list syntax.

### Expected behavior:
Name is shown correctly - purely text.

#### NVDA Installed/portable/running from source:
Installed

#### NVDA version:
Alpha.15781

#### Windows version:
Windows 10 Version 1803

#### Name and version of other software in use when reproducing the issue:
None

#### Other information about your system:
None

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA?
Yes (2018.2.1)

## Technical:
This is caused by the fact that, in old add-ons (ones generated by hand), manifest summary wasn't surrounded by quotes. This is the case where the string contained commas and other punctuation that could be interpreted by Python as declaring a list or other data structures. Unfortunatley, this fact isn't communicated to developers via developer guide.

Thus I propose that we modify the developer guide to include a note regarding this case, and to perhaps encourage people to use the add-on template, which will format strings like this correctly in the manifest.

Thanks.

### PR DESCRIPTION
**Pull request by josephsl**
### Link to issue number:
Fixes #8593

### Summary of the issue:
Clarify proper value types for add-on manifest fields.

### Description of how this pull request fixes the issue:
In add-on manifest section in developer guide, add value types for manifest fields. Also, added a note regarding enclosing summary and description data in quotes, and for multi-line descriptions, advised readers to use triple quotes.

### Testing performed:
Compiled developer guide.

### Known issues with pull request:
None

### Change log entry:
None

### CODE REVIEW

**Comment by jscholes** [comment]
In the example manifest, the author field isn't enclosed in quotes as you state it should be.

**Comment by josephsl** [comment]
Hi, right, I’ll address that in the next commit. Thanks.

**Review by derekriemer — DISMISSED** [review]
Make sure to quote all string fields. Also, see other comments.

**Comment by jscholes** [comment]
Agree with @derekriemer on this one.  For the most part the programming concept of a string calls for it to be quoted.  Usually, there is no such thing as a string which isn't quoted and the only thing allowing it in this case is the fact that the INI file format and parser don't care.  It would be more straight forward to just tell people to quote all strings in their manifest, including the internal add-on name.  If NVDA itself can't handle that, the abstraction is leaking.

At the very least, it would be clearer to remove the "enclosed in quotes" bit.  Have the type "string" in brackets as you have here, but if quoting is truly only appropriate for certain fields, write something like:

> This value must be enclosed in quotes."

At the end of the field description.

Example:

> summary (string): The name of the add-on as shown to the user. This value must be enclosed in quotes. (Mandatory)

Re: email addresses: presumably Mick originally wrote this and is happy for his address to be there, but it doesn't really matter either way.

**Comment by michaelDCurran** [comment]
If you are aiming for 2018.3 then the base branch must be now beta.

**Comment by josephsl** [comment]
Hi, shall be done. P.S. Can you let developers know when that happens so PR writers can rebase their work on beta please? Thanks.

**Comment by michaelDCurran** [comment]
This is on a case by case basis.

The developer guide is fine. All other PRs, unless they are fixing a 
regression after 2018.2.1, should be still based on master and will not 
be in 2018.3.

**Comment by josephsl** [comment]
Hi,

Note that for now, beta and master are the same, so when I do git rebase beta on my branch, it shows that it is up to date.

Thanks.

**Comment by michaelDCurran** [comment]
That is right. But you need to change the base branchin the pr itself on 
Github.

**Comment by josephsl** [comment]
Hi,

For future reference: to change the base branch, select edit title, select change base, then select the new base branch. When asked to confirm, say "change".

Thanks.

**Comment by josephsl** [comment]
Hi, done. Thanks.

File: developerGuide.t2t [hunk 1]
```diff
@@ -654,24 +654,25 @@ This manifest file contains key = value pares declaring info such as the add-on'
...
 Although it is highly suggested that manifests contain all fields, the fields marked as mandatory must be included.
 Otherwise, the add-on will not install.  
-- name: A short unique name for the add-on. This is used to differentiate add-ons internally and is not shown to the user. (Mandatory)
-- summary: The name of the add-on as shown to the user. (Mandatory)
-- version: The version of this add-on; e.g. 2.0. (Mandatory)
-- author: The author of this add-on, preferably in the form Full Name <email address>; e.g. Michael Curran <mick@kulgan.net>. (Mandatory)
-- description: A sentence or two describing what the add-on does.
-- url: A URL where this add-on, further info and upgrades can be found.
-- docFileName: The name of the main documentation file for this add-on; e.g. readme.html. See the [Add-on Documentation #AddonDoc] section for more details.
+- name (string enclosed in quotes): A short unique name for the add-on. This is used to differentiate add-ons internally and is not shown to the user. (Mandatory)
```
Comment (feerrenrut) [inline developerGuide.t2t]: I would prefer if this just said string. Then a note underneath: "strings should be enclosed in quotes, (see examples)"

**Review by feerrenrut — APPROVED** [review]
Thanks for addressing my concern about the repetition. The PR was targeting beta, I think it would be ok to merge this into into RC to be included in the 2019.1.1 release, but it will need to be rebased on RC very quickly.

**Comment by feerrenrut** [comment]
Actually, if this is delivered to RC it won't be translated. I will merge this to master now.

**Comment by lukaszgo1** [comment]
Sorry for going slightly offtopic here, but is Dev guide translated at all?

**Comment by Adriani90** [comment]
Yes it is, but I am not sure which version. I don't remember any revision being submited to the translation system during last 2 or 3 years. The version on the translation system might be outdated.

**Comment by josephsl** [comment]
Hi, the “dev guide” that folks will read in other languages is not this one – it is the add-on dev guide that has been translated. Thanks.

**Pull request merged**